src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -153,14 +155,24 @@ static MultiTable * MultiSubqueryPushdownTable(Query *subquery);
  * plan and adds a root node to top of it. The  original query is only used for subquery
  * pushdown planning.
  *
+ * In order to support external parameters for the queries where planning
+ * is done on the original query, we need to replace the external parameters
+ * manually. To achive that for subquery pushdown planning, we pass boundParams
+ * to this function.
+ *
+ * Note that although router planner uses original query, the above does not
+ * apply given that we send the parameters to the workers explicitly on the
+ * execution.
+ *
  * We also pass queryTree and plannerRestrictionContext to the planner. They
  * are primarily used to decide whether the subquery is safe to pushdown.
  * If not, it helps to produce meaningful error messages for subquery
  * pushdown planning.
  */
 MultiTreeRoot *
 MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
-					   PlannerRestrictionContext *plannerRestrictionContext)
+					   PlannerRestrictionContext *plannerRestrictionContext,
```

Comment:
I considered not passing `boundParams` here. However, that would make things more complicated things on the `CreateDistributedPlan()` function by adding the checks there as well. But, I'm open to suggestions here.